easybuild/easyblocks/j/java.py

```diff
@@ -36,13 +37,30 @@
 from easybuild.easyblocks.generic.packedbinary import PackedBinary
 from easybuild.tools.filetools import adjust_permissions, change_dir, copy_dir, copy_file, remove_dir
 from easybuild.tools.run import run_cmd
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import X86_64, AARCH64, get_cpu_architecture
 
 
 class EB_Java(PackedBinary):
     """Support for installing Java as a packed binary file (.tar.gz)
     Use the PackedBinary easyblock and set some extra paths.
     """
 
+    def __init__(self, *args, **kwargs):
+        """ Init the Java easyblock adding a new javaarch template var """
+        myarch = get_cpu_architecture()
+        if myarch == X86_64:
+            javaarch = 'x64'
+        elif myarch == AARCH64:
+            javaarch = 'aarch64'
+        else:
+            raise EasyBuildError("Architecture %s is not supported for Java on EasyBuild", myarch)
+
+        super(EB_Java, self).__init__(*args, **kwargs)
+
+        self.cfg.template_values['javaarch'] = javaarch
```

Comment:
In which context do you need this?

Reply:
This also came up in the conf call today (raised by @Micket), currently Java is a special case and uses a modulerc, not sure what the advantage is of adding templating since the existing easyconfigs require architecture versionsuffixes (and it doesn't look like this will change).